dao-voting/tests/harness.rs

```diff
@@ -0,0 +1,897 @@
+                    no_votes: asset_amount / 4,
+                    acceptance_percentage: 10,
+                    proposal_transaction: call_data,
+                    deadline: 15,
+                }
+            );
+        }
+    }
+
+    mod revert {
+        use super::*;
+
+        #[tokio::test]
+        #[should_panic]
+        async fn panics_on_not_enough_votes() {
+            let (_gov_token, gov_token_id, deployer, user, asset_amount) = setup().await;
+            deployer
+                .dao_voting
+                .constructor(gov_token_id)
+                .call()
+                .await
+                .unwrap()
+                .value;
+
+            let call_data = get_call_data(gov_token_id);
+
+            user.dao_voting
+                .create_proposal(10, 10, call_data)
+                .call()
+                .await
+                .unwrap()
+                .value;
+
+            user.dao_voting
+                .vote(true, 0, asset_amount / 4)
+                .call()
+                .await
+                .unwrap()
+                .value;
+        }
+
+        #[tokio::test]
+        #[should_panic]
+        async fn panics_on_expired_proposal() {
+            let (_gov_token, gov_token_id, deployer, user, asset_amount) = setup().await;
+            deployer
+                .dao_voting
+                .constructor(gov_token_id)
+                .call()
+                .await
+                .unwrap()
+                .value;
+
+            assert!(
+                deployer
+                    .gov_token
+                    .unwrap()
+                    .mint_and_send_to_address(100, user.wallet.address())
+                    .append_variable_outputs(1)
+                    .call()
+                    .await
+                    .unwrap()
+                    .value
+            );
+
+            let call_data = get_call_data(gov_token_id);
+
+            user.dao_voting
+                .create_proposal(1, 1, call_data)
+                .call()
+                .await
+                .unwrap()
+                .value;
+
+            let tx_params = TxParameters::new(None, Some(1_000_000), None, None);
+            let call_params = CallParameters::new(
+                Some(asset_amount),
+                Some(AssetId::from(*gov_token_id)),
+                Some(100_000),
+            );
+            user.dao_voting
+                .deposit()
+                .tx_params(tx_params)
+                .call_params(call_params)
+                .call()
+                .await
+                .unwrap()
+                .value;
+
+            user.dao_voting
+                .vote(true, 0, asset_amount / 4)
+                .call()
+                .await
+                .unwrap()
+                .value;
+        }
+    }
+}
+
+mod execute_proposal {
+    use super::*;
+
+    mod success {
+        use super::*;
+
+        #[tokio::test]
+        async fn user_proposal_can_execute() {
```

Comment:
you must at least call the execute function. if the VM does not support the functionality yet then comment out the test otherwise implement it fully before opening a PR. Basically no one is going to review things as thoroughly as me so they wouldn't spot this and problems could arise in master.

Reply:
I was unable to get the execute function to work.  I think its because I'm passing in dummy data which causes an invalid function call and a revert.  I'm going to comment out the test until I can properly encode a function selector and arguments.

Reply:
I think it is acceptable to either skeleton out as much of it as you can so that you have less work later (and comment it out) or just create the boilerplate to indicate which functional conditions need to be tested and then comment it all out. That way it's clear what needs to be tested later. There is an alternative where you should be able to annotate the test such that it automatically skips it but you'll need to look up what the annotation is to do that